recipes/libvault/all/patches/0.48.0-0001-fix-cmake.patch

```diff
@@ -0,0 +1,42 @@
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -1,10 +1,6 @@
+ cmake_minimum_required(VERSION 3.12)
+ project(vault VERSION 0.48.0 DESCRIPTION "Vault library for C++")
+ 
+-set(CMAKE_CXX_STANDARD 17)
+-set(CMAKE_CXX_STANDARD_REQUIRED ON)
+-set(CMAKE_CXX_EXTENSIONS OFF)
+-set(CMAKE_CXX_FLAGS_DEBUG --coverage)
+ set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+ 
+ option(ENABLE_TEST "Enable tests?" ON)
+@@ -14,12 +10,7 @@ option(LINK_CURL "Link curl library for vault" OFF)
+ option(BUILD_SHARED_LIBS "Build vault as a shared library" ON)
+ option(INSTALL "Run install targets" ON)
+ 
+-find_package(CURL)
+-if (CURL_FOUND)
+-  include_directories(${CURL_INCLUDE_DIR})
+-else (CURL_FOUND)
+-  message(FATAL_ERROR "CURL not found")
+-endif (CURL_FOUND)
++find_package(CURL REQUIRED)
+ 
+ include(GNUInstallDirs)
+ include_directories("${PROJECT_SOURCE_DIR}/lib")
+@@ -120,9 +111,11 @@ set_target_properties(vault PROPERTIES
+ 
+ target_include_directories(vault PRIVATE src)
+ 
+-if(LINK_CURL)
+-  target_link_libraries(vault curl)
+-endif(LINK_CURL)
++target_link_libraries(vault PUBLIC CURL::libcurl)
++target_compile_features(vault PUBLIC cxx_std_17)
```

Comment:
As this is for `vault`, does this not just override whatever I set in my `compiler.cppstd` setting?

Reply:
`CMAKE_CXX_STANDARD` injected by `CMakeToolchain` when `compiler.cppstd` is set has precedence over cxx_std compile feature.

`compiler.cppstd` is not honored when upstream CMakeLists hardcode `CMAKE_CXX_STANDARD`. The modern approach for CMake is "compile features" in libraries to express requirements, and CMAKE_CXX_STANDARD externally to force a specific standard if needed.

Reply:
Thank you for your explanation, it could be pushed to the upstream too.